orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -4217,9 +4217,12 @@ def test_ortmodule_string_inputs_are_ignored(caplog):
 
     target_str = "Received input of type <class 'str'> which may be treated as a constant by ORT by default."
     found_target_str = False
-    for record in caplog.records:
-        if target_str in record.message:
-            found_target_str = True
+    with warnings.catch_warnings(record=True) as w:
+        for i in range(len(w)):
```

Comment:
Looks like i is only used for indexing? Prefer iterating on w instead.

found_target_str = any(target_str in warning.message for warning in w)

Reply:
if I remember correctly, w is not iteratable, so I used this loop.

Reply:
Why would an indexable not be an iterable? Do you know its type?